torchvision/utils.py

```diff
@@ -300,6 +301,70 @@ def draw_segmentation_masks(
     return out.to(out_dtype)
 
 
+@torch.no_grad()
+def draw_keypoints(
+    image: torch.Tensor,
+    keypoints: torch.Tensor,
+    labels: Optional[List[str]] = None,
```

Comment:
Instead of providing `labels` (which, for the COCO dataset that we consider we only have one category), I would instead provide a `connectivity` argument, which dictates which points should be linked against which other point.

This can be an optional argument, and if provided will connect the different keypoints together (and could thus replace the `connect` boolean argument as well.

Here is an implementation that I had [for it for the "person" class](https://github.com/facebookresearch/maskrcnn-benchmark/blob/57eec25b75144d9fb1a6857f32553e1574177daf/demo/predictor.py#L413-L471)

Reply:
Hi @fmassa . Thanks a lot for review :smiley: 

The idea behind labels is if person wants to explicitly mark the keypoint number.
Particularly useful in counting the keypoints, or validating performance of all the keypoints. (Maybe a keypoint number X is performing very poorly? User would need to label these to come to know.

Also it might be particularly useful in face detection etc models. 

![image](https://user-images.githubusercontent.com/47158509/132563157-4a43b22a-6da3-4168-aeb9-9329f9232367.png)

Maybe users want to do something similar to above. 

About the connectivity point, I think you are right. connectivity can be `Optional[Tuple[Tuple[int, int]]]` denoting which two points should be connected. So that we can link them up if provided.

I think the utility should be slightly generic and not limited to COCO. Let me know what you think

The person implementation is very nice. (maskrcnn-benchmark is probably one of the best libraries)
I think I can post images side by side how maskrcnn implementation and the current one (when finalized a bit more) looks like.

Reply:
I think for keypoints (given that they are fairly small) labelling them might not be very readable, while using a color-code might be easier to spot. This begs the question if the color should represent the instance or the keypoint-id in the instance.

My 2 cents is that we can pass a color tensor that is of size `(num_instances, num_keypoints, 3)`, and that internally we can broadcast to the full shape if the user only passed `(num_keypoints, 3)` or `(num_instances, 1, 3)` or `(3)`.

Thoughts?

Reply:
Yes, we do not use labels too for drawing segmentation masks. And a good color combination can be sufficient for visualizing keypoints. 

We don't use tensors for colors, in other utlitites so maybe we shouldn't use here.
I suggest colors to be either "str" (One color for all keypoints) List[str] specifying color for each the keypoints ids. 
None, so that we generate random colors from palette. 

Specifying color per instance might be hard and users would need to dynamically adjust this utlity, as number of detected instances would differ. We don't use num_instances in any other utility and would prefer to stick with keypoint ids.